### ISSUE TITLE
RPC client error: sendRequest failed: websocket routine exiting

### ISSUE THREAD

**Issue by hyunmoon** [filecoin-project/lotus#4063]
**Describe the bug**
This issue is about the connection between lotus-miner and lotus-worker.

websocket i/o timeout after completing PreCommit2 causes the following error repeatedly:
```
ERROR	advmgr	sector-storage/sched.go:370	trySched(1) req.sel.Ok error: getting supported worker task types:
    github.com/filecoin-project/lotus/extern/sector-storage.(*taskSelector).Ok
        /home/downloads/lotus/extern/sector-storage/selector_task.go:25
  - RPC client error: sendRequest failed: websocket routine exiting
```
- No errors on lotus-worker's side.
- No known hardware issues.
- Occurs on random servers.
- Recovery requires restarting the server.
- Not a one time incident. It's been happening dozens of times everyday since SR1.

**To Reproduce**
- Wait until a worker goes idle (after completing PreCommit2 and waiting for seed for example)
- Get the following errors:
![req_sel_ok_error](https://user-images.githubusercontent.com/2242149/94373003-35ab8c80-013d-11eb-90d8-7fcc71c5437d.JPG)

I've once been spammed by this error (over 10,000 identical logs occurred consecutively).
When it happened, lotus-miner went idle and missed windowed posts.
I had to restart lotus-miner to get it working again.

**Version (run `lotus version`):**
Daemon:  0.8.0+git.2c1d96bc+api0.15.0
Local: lotus-miner version 0.8.0+git.2c1d96bc

**Additional context**
API timeout has been increased from 30s (default) to 60s in miner's config.

**Comment by keesvanbochove** [comment]
I have this issue repeatedly too, the lotus daemon seems fine, and is correctly listening on port 1234 of its assigned IP address. Yet, lotus miners and workers eventually run into this error after a number of hours, both locally on the same server as where the daemon is running as well as on other servers.

Daemon:  0.8.0+git.2c1d96bc.dirty+api0.16.0
Local: lotus-miner version 0.8.0+git.2c1d96bc.dirty

`2020-09-27T22:02:44.748+0200	ERROR	rpc	go-jsonrpc@v0.1.2-0.20200822201400-474f4fdccc52/websocket.go:654	Connection timeout	{"remote": "192.168.1.252:1234"}
2020-09-27T22:02:44.748+0200	WARN	storageminer	storage/wdpost_sched.go:112	window post scheduler notifs channel closed
2020-09-27T22:02:44.748+0200	WARN	events	events/events.go:97	listenHeadChanges quit
2020-09-27T22:02:44.748+0200	ERROR	storageminer	storage/wdpost_sched.go:100	ChainNotify error: RPC client error: sendRequest failed: websocket routine exiting
2020-09-27T22:02:44.748+0200	WARN	events	events/events.go:97	listenHeadChanges quit
2020-09-27T22:02:44.748+0200	WARN	rpc	go-jsonrpc@v0.1.2-0.20200822201400-474f4fdccc52/websocket.go:289	failed to send request	{"method": "xrpc.cancel", "id": 8, "error": "write tcp 192.168.1.223:57574->192.168.1.252:1234: use of closed network connection"}
2020-09-27T22:02:44.748+0200	WARN	rpc	go-jsonrpc@v0.1.2-0.20200822201400-474f4fdccc52/websocket.go:289	failed to send request	{"method": "xrpc.cancel", "id": 20, "error": "write tcp 192.168.1.223:57574->192.168.1.252:1234: use of closed network connection"}
2020-09-27T22:02:44.750+0200	ERROR	miner	miner/miner.go:226	mining block failed: failed to get mining base info:
    github.com/filecoin-project/lotus/miner.(*Miner).mineOne
        /home/kees/projects/lotus/miner/miner.go:360
  - handler: websocket connection closed
2020-09-27T22:02:45.748+0200	INFO	events	events/events.go:104	restarting listenHeadChanges
2020-09-27T22:02:45.748+0200	INFO	events	events/events.go:104	restarting listenHeadChanges
2020-09-27T22:02:45.748+0200	ERROR	events	events/events.go:95	listen head changes errored: listenHeadChanges ChainNotify call failed: RPC client error: sendRequest failed: websocket routine exiting
2020-09-27T22:02:45.748+0200	ERROR	events	events/events.go:95	listen head changes errored: listenHeadChanges ChainNotify call failed: RPC client error: sendRequest failed: websocket routine exiting
2020-09-27T22:02:45.751+0200	ERROR	miner	miner/miner.go:172	failed to get best mining candidate: RPC client error: sendRequest failed: websocket routine exiting
2020-09-27T22:02:46.748+0200	INFO	events	events/events.go:104	restarting listenHeadChanges
2020-09-27T22:02:46.748+0200	INFO	events	events/events.go:104	restarting listenHeadChanges
2020-09-27T22:02:46.748+0200	ERROR	events	events/events.go:95	listen head changes errored: listenHeadChanges ChainNotify call failed: RPC client error: sendRequest failed: websocket routine exiting
2020-09-27T22:02:46.748+0200	ERROR	events	events/events.go:95	listen head changes errored: listenHeadChanges ChainNotify call failed: RPC client error: sendRequest failed: websocket routine exiting
2020-09-27T22:02:47.748+0200	INFO	events	events/events.go:104	restarting listenHeadChanges
2020-09-27T22:02:47.749+0200	INFO	events	events/events.go:104	restarting listenHeadChanges
`
etc.

**Comment by jennijuju** [comment]
@hyunmoon @keesvanbochove Is the port it tries to connects to the right port you set in the config? (want to see if this is related to #3773 )

**Comment by hyunmoon** [comment]
@jennijuju Yes. As you can see in the screenshot I posted, the miner (192.168.10.4) tries to connect to the worker (192.168.10.173:2222). 2222 is the listening port I set on the worker.

**Comment by damonhung** [comment]
I think I just hit this also.  I didn't even notice anything wrong until I saw myself totally faulted.  Mine started with this:

```2020-09-27T13:47:14.644-0400    DEBUG        stores  stores/local.go:559     moving {9737 93}(4) to storage: 11a563f1-9d02-4e94-9ff8-1ae02d35e289(se:true; st:false) -> 4d0061ef-c7dd-42e4-b82d-02fae6e45707(se:false; st:true)
2020-09-27T13:47:14.649-0400    DEBUG        stores  stores/util_unix.go:28  move sector data        {"from": "/scratch/cache/s-t09737-93", "to": "/data/FileCoin/cache/s-t09737-93"}
2020-09-27T13:47:14.940-0400    INFO sectors storage-sealing/states_sealing.go:465   Proving sector 93
2020-09-27T13:47:14.940-0400    WARN advmgr  sector-storage/manager.go:468   ReleaseUnsealed todo
2020-09-27T13:47:17.155-0400    ERROR        rpc     go-jsonrpc@v0.1.2-0.20200822201400-474f4fdccc52/websocket.go:654        Connection timeout      {"remote": "127.0.0.1:1234"}
2020-09-27T13:47:17.254-0400    WARN storageminer    storage/wdpost_sched.go:112     window post scheduler notifs channel closed
2020-09-27T13:47:17.371-0400    ERROR        events  events/events_called.go:382     event diff fn failed: handler: websocket connection closed
2020-09-27T13:47:17.373-0400    ERROR        events  events/events_called.go:382     event diff fn failed: RPC client error: sendRequest failed: websocket routine exiting
2020-09-27T13:47:17.373-0400    ERROR        events  events/events_called.go:382     event diff fn failed: RPC client error: sendRequest failed: websocket routine exiting
2020-09-27T13:47:17.373-0400    ERROR        events  events/events_called.go:382     event diff fn failed: RPC client error: sendRequest failed: websocket routine exiting
```

The `event diff fn failed` message repeats 1060 more times between 13:47:17.373-0400 and 13:47:18.612-0400.  Filtering those out leaves:

```
2020-09-27T13:47:17.384-0400    ERROR   miner   miner/miner.go:226      mining block failed: failed to get mining base info:
    github.com/filecoin-project/lotus/miner.(*Miner).mineOne
        /home/moo/lotus/miner/miner.go:360
  - handler: websocket connection closed
2020-09-27T13:47:17.385-0400    ERROR   events  events/events_called.go:526     messagesForTs MessagesForBlock failed (ts.H=97410, Bcid:bafy2bzacebg7j3h66ebw6wolnjg4nfbgniedff6yw2kihopliw7rawfhtgnwe, B.Mcid:bafy2bzaceayhlqwqsm22otds5k2gwvprc36eclhuznoeyvlkgi4rwcclyhmjm): handler: websocket connection closed
2020-09-27T13:47:17.412-0400    ERROR   events  events/events_called.go:526     messagesForTs MessagesForBlock failed (ts.H=97410, Bcid:bafy2bzaceayekplags6kn3im7ejxnlu4w5borvoynelsu46mnnwpwfq7gzf6o, B.Mcid:bafy2bzacecmda75ovposbdateg7eyhwij65zklgyijgcjwynlklmqazpwlhba): RPC client error: sendRequest failed: websocket routine exiting
2020-09-27T13:47:17.441-0400    ERROR   events  events/events_called.go:478     getting parent tipset in checkNewCalls: RPC client error: sendRequest failed: websocket routine exiting
2020-09-27T13:47:17.441-0400    WARN    events  events/events.go:155    headChange failed: RPC client error: sendRequest failed: websocket routine exiting
2020-09-27T13:47:17.441-0400    WARN    events  events/events.go:97     listenHeadChanges quit
2020-09-27T13:47:17.465-0400    ERROR   storageminer    storage/wdpost_sched.go:100     ChainNotify error: RPC client error: sendRequest failed: websocket routine exiting
2020-09-27T13:47:17.532-0400    WARN    rpc     go-jsonrpc@v0.1.2-0.20200822201400-474f4fdccc52/websocket.go:289        failed to send request  {"method": "xrpc.cancel", "id": 25, "error": "write tcp 127.0.0.1:37636->127.0.0.1:1234: use of closed network connection"}
2020-09-27T13:47:17.664-0400    ERROR   events  events/events_called.go:478     getting parent tipset in checkNewCalls: RPC client error: sendRequest failed: websocket routine exiting
2020-09-27T13:47:17.664-0400    WARN    events  events/events.go:155    headChange failed: RPC client error: sendRequest failed: websocket routine exiting
2020-09-27T13:47:18.028-0400    ERROR   events  events/events_called.go:478     getting parent tipset in checkNewCalls: RPC client error: sendRequest failed: websocket routine exiting
2020-09-27T13:47:18.028-0400    WARN    events  events/events.go:155    headChange failed: RPC client error: sendRequest failed: websocket routine exiting
2020-09-27T13:47:18.321-0400    ERROR   events  events/events_called.go:478     getting parent tipset in checkNewCalls: RPC client error: sendRequest failed: websocket routine exiting
2020-09-27T13:47:18.321-0400    WARN    events  events/events.go:155    headChange failed: RPC client error: sendRequest failed: websocket routine exiting
2020-09-27T13:47:18.385-0400    ERROR   miner   miner/miner.go:172      failed to get best mining candidate: RPC client error: sendRequest failed: websocket routine exiting
2020-09-27T13:47:18.442-0400    INFO    events  events/events.go:104    restarting listenHeadChanges
2020-09-27T13:47:18.458-0400    ERROR   events  events/events.go:95     listen head changes errored: listenHeadChanges ChainNotify call failed: RPC client error: sendRequest failed: websocket routine exiting
2020-09-27T13:47:18.612-0400    ERROR   events  events/events_called.go:478     getting parent tipset in checkNewCalls: RPC client error: sendRequest failed: websocket routine exiting
2020-09-27T13:47:18.612-0400    WARN    events  events/events.go:155    headChange failed: RPC client error: sendRequest failed: websocket routine exiting
2020-09-27T13:47:18.612-0400    WARN    events  events/events.go:97     listenHeadChanges quit
2020-09-27T13:47:18.612-0400    WARN    rpc     go-jsonrpc@v0.1.2-0.20200822201400-474f4fdccc52/websocket.go:289        failed to send request  {"method": "xrpc.cancel", "id": 8, "error": "write tcp 127.0.0.1:37636->127.0.0.1:1234: use of closed network connection"}
2020-09-27T13:47:19.458-0400    INFO    events  events/events.go:104    restarting listenHeadChanges
2020-09-27T13:47:19.503-0400    ERROR   events  events/events.go:95     listen head changes errored: listenHeadChanges ChainNotify call failed: RPC client error: sendRequest failed: websocket routine exiting
2020-09-27T13:47:19.612-0400    INFO    events  events/events.go:104    restarting listenHeadChanges
2020-09-27T13:47:19.618-0400    ERROR   events  events/events.go:95     listen head changes errored: listenHeadChanges ChainNotify call failed: RPC client error: sendRequest failed: websocket routine exiting
2020-09-27T13:47:20.503-0400    INFO    events  events/events.go:104    restarting listenHeadChanges
2020-09-27T13:47:20.523-0400    ERROR   events  events/events.go:95     listen head changes errored: listenHeadChanges ChainNotify call failed: RPC client error: sendRequest failed: websocket routine exiting
```

The `restarting listenHeadChanges` / `listen head changes errored` messages repeat twice per second forever.  Once every 5 seconds, we also have

`2020-09-27T13:47:23.482-0400    ERROR   miner   miner/miner.go:172      failed to get best mining candidate: RPC client error: sendRequest failed: websocket routine exiting`

And once every 10 seconds, we have

`2020-09-27T13:47:27.476-0400    ERROR   storageminer    storage/wdpost_sched.go:100     ChainNotify error: RPC client error: sendRequest failed: websocket routine exiting`

This went on for many hours.  Restarting `lotus-miner` seems to have fixed it.

**Comment by jennijuju** [comment]
related #3830 , #3940

**Comment by damonhung** [comment]
Had another instance today, this time running v0.8.1.  Looks a little bit different, but then goes into the same repetition of `restarting listenHeadChanges` / `listen head changes errored`.

```2020-10-05T10:01:02.054-0400    DEBUG   advmgr  sector-storage/sched.go:324     SCHED 1 queued; 2 open windows
2020-10-05T10:01:02.054-0400    DEBUG   advmgr  sector-storage/sched.go:415     SCHED windows: [{allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>} todo:[]} {allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>} todo:[]}]
2020-10-05T10:01:02.054-0400    DEBUG   advmgr  sector-storage/sched.go:416     SCHED Acceptable win: [[0 1]]
2020-10-05T10:01:02.054-0400    DEBUG   advmgr  sector-storage/sched.go:430     SCHED try assign sqi:0 sector 110 to window 0
2020-10-05T10:01:02.054-0400    DEBUG   advmgr  sector-storage/sched.go:437     SCHED ASSIGNED sqi:0 sector 110 task seal/v0/commit/2 to window 0
2020-10-05T10:01:02.054-0400    DEBUG   advmgr  sector-storage/sched.go:596     assign worker sector 110
2020-10-05T10:01:02.054-0400    DEBUG   advmgr  sector-storage/sched.go:324     SCHED 0 queued; 2 open windows
2020-10-05T10:01:02.054-0400    DEBUG   advmgr  sector-storage/sched.go:415     SCHED windows: [{allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>} todo:[]} {allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>} todo:[]}]
2020-10-05T10:01:02.054-0400    DEBUG   advmgr  sector-storage/sched.go:416     SCHED Acceptable win: []
2020-10-05T10:01:02.054-0400    DEBUG   advmgr  sector-storage/sched.go:557     task done       {"workerid": 0}
2020-10-05T10:01:02.060 INFO filcrypto::proofs::api > seal_commit_phase2: start
2020-10-05T10:01:02.249-0400    INFO    rpc     go-jsonrpc@v0.1.2-0.20200822201400-474f4fdccc52/client.go:213   rpc output message buffer       {"n": 2}
2020-10-05T10:01:02.268-0400    INFO    rpc     go-jsonrpc@v0.1.2-0.20200822201400-474f4fdccc52/client.go:213   rpc output message buffer       {"n": 3}
2020-10-05T10:01:02.291 INFO filecoin_proofs::api::seal > seal_commit_phase2:start
2020-10-05T10:01:02.292 INFO filecoin_proofs::caches > trying parameters memory cache for: STACKED[34359738368]
2020-10-05T10:01:02.292 INFO filecoin_proofs::caches > no params in memory cache for STACKED[34359738368]
2020-10-05T10:01:02.294 INFO storage_proofs_core::parameter_cache > parameter set identifier for cache: layered_drgporep::PublicParams{ graph: stacked_graph::StackedGraph{expansion_degree: 8 base_graph: drgraph::BucketGraph{size: 1073741824; degree: 6; hasher: poseidon_hasher} }, challenges: LayerChallenges { layers: 11, max_count: 18 }, tree: merkletree-poseidon_hasher-8-8-0 }
2020-10-05T10:01:02.294 INFO storage_proofs_core::parameter_cache > ensuring that all ancestor directories for: "/var/tmp/filecoin-proof-parameters/v28-stacked-proof-of-replication-merkletree-poseidon_hasher-8-8-0-sha256_hasher-82a357d2f2ca81dc61bb45f4a762807aedee1b0a53fd6c4e77b46a01bfef7820.params" exist
2020-10-05T10:01:02.294 INFO storage_proofs_core::parameter_cache > checking cache_path: "/var/tmp/filecoin-proof-parameters/v28-stacked-proof-of-replication-merkletree-poseidon_hasher-8-8-0-sha256_hasher-82a357d2f2ca81dc61bb45f4a762807aedee1b0a53fd6c4e77b46a01bfef7820.params" for parameters
2020-10-05T10:01:04.723-0400    INFO    rpc     go-jsonrpc@v0.1.2-0.20200822201400-474f4fdccc52/client.go:213   rpc output message buffer       {"n": 2}
2020-10-05T10:01:04.807 INFO storage_proofs_core::parameter_cache > read parameters from cache "/var/tmp/filecoin-proof-parameters/v28-stacked-proof-of-replication-merkletree-poseidon_hasher-8-8-0-sha256_hasher-82a357d2f2ca81dc61bb45f4a762807aedee1b0a53fd6c4e77b46a01bfef7820.params"
2020-10-05T10:01:04.807 INFO filecoin_proofs::api::seal > got groth params (34359738368) while sealing
2020-10-05T10:01:04.807 INFO filecoin_proofs::api::seal > snark_proof:start
2020-10-05T10:01:04.823 INFO bellperson::groth16::prover > Bellperson 0.9.2 is being used!
2020-10-05T10:01:32.922-0400    INFO    rpc     go-jsonrpc@v0.1.2-0.20200822201400-474f4fdccc52/client.go:213   rpc output message buffer       {"n": 2}
2020-10-05T10:01:32.941-0400    INFO    rpc     go-jsonrpc@v0.1.2-0.20200822201400-474f4fdccc52/client.go:213   rpc output message buffer       {"n": 3}
2020-10-05T10:02:31.020-0400    INFO    rpc     go-jsonrpc@v0.1.2-0.20200822201400-474f4fdccc52/client.go:213   rpc output message buffer       {"n": 2}
2020-10-05T10:02:31.037-0400    INFO    rpc     go-jsonrpc@v0.1.2-0.20200822201400-474f4fdccc52/client.go:213   rpc output message buffer       {"n": 3}
2020-10-05T10:02:35.284-0400    INFO    rpc     go-jsonrpc@v0.1.2-0.20200822201400-474f4fdccc52/client.go:213   rpc output message buffer       {"n": 3}
2020-10-05T10:03:00.477-0400    INFO    rpc     go-jsonrpc@v0.1.2-0.20200822201400-474f4fdccc52/client.go:213   rpc output message buffer       {"n": 2}
2020-10-05T10:03:00.477-0400    INFO    rpc     go-jsonrpc@v0.1.2-0.20200822201400-474f4fdccc52/client.go:213   rpc output message buffer       {"n": 2}
2020-10-05T10:03:00.758-0400    INFO    rpc     go-jsonrpc@v0.1.2-0.20200822201400-474f4fdccc52/client.go:213   rpc output message buffer       {"n": 3}
2020-10-05T10:03:05.462-0400    INFO    rpc     go-jsonrpc@v0.1.2-0.20200822201400-474f4fdccc52/client.go:213   rpc output message buffer       {"n": 3}
2020-10-05T10:03:05.495-0400    INFO    rpc     go-jsonrpc@v0.1.2-0.20200822201400-474f4fdccc52/client.go:213   rpc output message buffer       {"n": 4}
2020-10-05T10:03:30.907-0400    INFO    rpc     go-jsonrpc@v0.1.2-0.20200822201400-474f4fdccc52/client.go:213   rpc output message buffer       {"n": 2}
2020-10-05T10:06:13.492-0400    ERROR   rpc     go-jsonrpc@v0.1.2-0.20200822201400-474f4fdccc52/websocket.go:654        Connection timeout      {"remote": "127.0.0.1:1234"}
2020-10-05T10:06:13.821-0400    WARN    events  events/events.go:97     listenHeadChanges quit
2020-10-05T10:06:13.821-0400    WARN    events  events/events.go:97     listenHeadChanges quit
2020-10-05T10:06:13.831-0400    WARN    storageminer    storage/wdpost_sched.go:106     window post scheduler notifs channel closed
2020-10-05T10:06:13.847-0400    WARN    rpc     go-jsonrpc@v0.1.2-0.20200822201400-474f4fdccc52/websocket.go:289        failed to send request  {"method": "xrpc.cancel", "id": 20, "error": "write tcp 127.0.0.1:55450->127.0.0.1:1234: use of closed network connection"}
2020-10-05T10:06:13.848-0400    WARN    rpc     go-jsonrpc@v0.1.2-0.20200822201400-474f4fdccc52/websocket.go:289        failed to send request  {"method": "xrpc.cancel", "id": 8, "error": "write tcp 127.0.0.1:55450->127.0.0.1:1234: use of closed network connection"}
2020-10-05T10:06:13.886-0400    ERROR   miner   miner/miner.go:226      mining block failed: failed to get mining base info:
    github.com/filecoin-project/lotus/miner.(*Miner).mineOne
        /home/moo/lotus/miner/miner.go:360
  - handler: websocket connection closed
2020-10-05T10:06:13.886-0400    ERROR   storageminer    storage/wdpost_sched.go:94      ChainNotify error: RPC client error: sendRequest failed: websocket routine exiting
2020-10-05T10:06:14.834-0400    INFO    events  events/events.go:104    restarting listenHeadChanges
2020-10-05T10:06:14.834-0400    INFO    events  events/events.go:104    restarting listenHeadChanges
2020-10-05T10:06:14.834-0400    ERROR   events  events/events.go:95     listen head changes errored: listenHeadChanges ChainNotify call failed: RPC client error: sendRequest failed: websocket routine exiting
2020-10-05T10:06:14.851-0400    ERROR   events  events/events.go:95     listen head changes errored: listenHeadChanges ChainNotify call failed: RPC client error: sendRequest failed: websocket routine exiting
2020-10-05T10:06:14.886-0400    ERROR   miner   miner/miner.go:172      failed to get best mining candidate: RPC client error: sendRequest failed: websocket routine exiting
```

Another difference in v0.8.1 is that, once the miner is in this state, I see this once every 20 minutes.  (I've snipped out the long proofs.)

```2020-10-05T10:52:42.735-0400    WARN rpc     go-jsonrpc@v0.1.2-0.20200822201400-474f4fdccc52/handler.go:241  error in RPC call to 'Filecoin.SectorsStatus': RPC client error: sendRequest failed: websocket routine exiting
2020-10-05T10:52:42.735-0400    ERROR        rpc     go-jsonrpc@v0.1.2-0.20200822201400-474f4fdccc52/handler.go:283  error and res returned  {"request": {"jsonrpc":"2.0","id":3,"method":"Filecoin.SectorsStatus","params":[0,true],"meta":{"SpanContext":"AACJA56Yhb/sB14WlCno/YieAZAOawImjfg3AgA="}}, "r.err": "RPC client error: sendRequest failed: websocket routine exiting", "res": {"SectorID":0,"State":"Proving","CommD":{"/":"baga6ea4seaqao7s73y24kcutaosvacpdjgfe5pw76ooefnyqw4ynr3d2y6x2mpq"},"CommR":{"/":"bagboea4b5abcas6s3tsgq6va5a5ixmi4rs2n32axelfshjlu4unhbmvlv3dcaeic"},"Proof":"<SNIP>","Deals":[0],"Ticket":{"Value":"b+syukRKA48HpjKejgm4IQV7FT4yJntIMF9ZcK3R/zs=","Epoch":6791},"Seed":{"Value":"6cr12Ke89vs/WgItOazVWBVY1kTrExBGgmSzevMM2Ho=","Epoch":8434},"PreCommitMsg":{"/":"bafy2bzacedmbrenclnomaxnm24oeka6pryxh6hyr2adlz4lv6a2w7etv35y2g"},"CommitMsg":{"/":"bafy2bzaceaasb64obtbjg2j2jjfmomjtuu5g2ilmsilrx6iiodi7ggrcdwkhu"},"Retries":0,"ToUpgrade":false,"LastErr":"","Log":[{"Kind":"event;sealing.SectorStartCC","Timestamp":1598537151,"Trace":"","Message":"{\"User\":{\"ID\":0,\"SectorType\":3,\"Pieces\":[{\"Piece\":{\"Size\":34359738368,\"PieceCID\":{\"/\":\"baga6ea4seaqao7s73y24kcutaosvacpdjgfe5pw76ooefnyqw4ynr3d2y6x2mpq\"}},\"DealInfo\":null}]}}"},{"Kind":"event;sealing.SectorPacked","Timestamp":1598537151,"Trace":"","Message":"{\"User\":{\"FillerPieces\":null}}"},{"Kind":"event;sealing.SectorPreCommit1","Timestamp":1598551742,"Trace":"","Message":"{\"User\":{\"PreCommit1Out\":\"<SNIP>\",\"TicketValue\":\"b+syukRKA48HpjKejgm4IQV7FT4yJntIMF9ZcK3R/zs=\",\"TicketEpoch\":6791}}"},{"Kind":"event;sealing.SectorPreCommit2","Timestamp":1598554894,"Trace":"","Message":"{\"User\":{\"Sealed\":{\"/\":\"bagboea4b5abcas6s3tsgq6va5a5ixmi4rs2n32axelfshjlu4unhbmvlv3dcaeic\"},\"Unsealed\":{\"/\":\"baga6ea4seaqao7s73y24kcutaosvacpdjgfe5pw76ooefnyqw4ynr3d2y6x2mpq\"}}}"},{"Kind":"event;sealing.SectorPreCommitted","Timestamp":1598554894,"Trace":"","Message":"{\"User\":{\"Message\":{\"/\":\"bafy2bzacedmbrenclnomaxnm24oeka6pryxh6hyr2adlz4lv6a2w7etv35y2g\"},\"PreCommitDeposit\":\"920493802844767324\",\"PreCommitInfo\":{\"SealProof\":3,\"SectorNumber\":0,\"SealedCID\":{\"/\":\"bagboea4b5abcas6s3tsgq6va5a5ixmi4rs2n32axelfshjlu4unhbmvlv3dcaeic\"},\"SealRandEpoch\":6791,\"DealIDs\":[],\"Expiration\":1558124,\"ReplaceCapacity\":false,\"ReplaceSectorDeadline\":0,\"ReplaceSectorPartition\":0,\"ReplaceSectorNumber\":0}}}"},{"Kind":"event;sealing.SectorPreCommitLanded","Timestamp":1598555099,"Trace":"","Message":"{\"User\":{\"TipSet\":\"AXGg5AIg9ZcMZJCS59koZk047aNvtpduUPVxUft6WH7hBv4fgKgBcaDkAiBHaa4BDll192dj47v3HUKn5XkZVUkeVMTlJ4nHFWJVYgFxoOQCIF+5kMNfofaF44baWACY5M1+LgsA0MNTgjDmWvNtGn+CAXGg5AIgeNyYyEAXCs12dxXamZrhTLSEPBL0/rkpWyKpJsDZItABcaDkAiAPkbwXWGujFu6dECbvXGzCsIvJF8tXAUHbgq6F6oqhyg==\"}}"},{"Kind":"event;sealing.SectorSeedReady","Timestamp":1598559605,"Trace":"","Message":"{\"User\":{\"SeedValue\":\"6cr12Ke89vs/WgItOazVWBVY1kTrExBGgmSzevMM2Ho=\",\"SeedEpoch\":8434}}"},{"Kind":"event;sealing.SectorCommitted","Timestamp":1598565009,"Trace":"","Message":"{\"User\":{\"Message\":{\"/\":\"bafy2bzaceaasb64obtbjg2j2jjfmomjtuu5g2ilmsilrx6iiodi7ggrcdwkhu\"},\"Proof\":\"<SNIP>\"}}"},{"Kind":"event;sealing.SectorProving","Timestamp":1598565210,"Trace":"","Message":"{\"User\":{}}"},{"Kind":"event;sealing.SectorFinalized","Timestamp":1598565284,"Trace":"","Message":"{\"User\":{}}"},{"Kind":"event;sealing.SectorRestart","Timestamp":1598604411,"Trace":"","Message":"{\"User\":{}}"},{"Kind":"event;sealing.SectorRestart","Timestamp":1598680495,"Trace":"","Message":"{\"User\":{}}"},{"Kind":"event;sealing.SectorRestart","Timestamp":1598732749,"Trace":"","Message":"{\"User\":{}}"},{"Kind":"event;sealing.SectorRestart","Timestamp":1598744489,"Trace":"","Message":"{\"User\":{}}"},{"Kind":"event;sealing.SectorRestart","Timestamp":1598745142,"Trace":"","Message":"{\"User\":{}}"},{"Kind":"event;sealing.SectorRestart","Timestamp":1598922732,"Trace":"","Message":"{\"User\":{}}"},{"Kind":"event;sealing.SectorRestart","Timestamp":1598923446,"Trace":"","Message":"{\"User\":{}}"},{"Kind":"event;sealing.SectorRestart","Timestamp":1598935528,"Trace":"","Message":"{\"User\":{}}"},{"Kind":"event;sealing.SectorRestart","Timestamp":1598939923,"Trace":"","Message":"{\"User\":{}}"},{"Kind":"event;sealing.SectorRestart","Timestamp":1599099879,"Trace":"","Message":"{\"User\":{}}"},{"Kind":"event;sealing.SectorRestart","Timestamp":1599206192,"Trace":"","Message":"{\"User\":{}}"},{"Kind":"event;sealing.SectorRestart","Timestamp":1599546228,"Trace":"","Message":"{\"User\":{}}"},{"Kind":"event;sealing.SectorRestart","Timestamp":1599733755,"Trace":"","Message":"{\"User\":{}}"},{"Kind":"event;sealing.SectorRestart","Timestamp":1599870297,"Trace":"","Message":"{\"User\":{}}"},{"Kind":"event;sealing.SectorRestart","Timestamp":1600404557,"Trace":"","Message":"{\"User\":{}}"},{"Kind":"event;sealing.SectorRestart","Timestamp":1601106891,"Trace":"","Message":"{\"User\":{}}"},{"Kind":"event;sealing.SectorRestart","Timestamp":1601274394,"Trace":"","Message":"{\"User\":{}}"},{"Kind":"event;sealing.SectorRestart","Timestamp":1601436641,"Trace":"","Message":"{\"User\":{}}"},{"Kind":"event;sealing.SectorRestart","Timestamp":1601746736,"Trace":"","Message":"{\"User\":{}}"},{"Kind":"event;sealing.SectorRestart","Timestamp":1601851345,"Trace":"","Message":"{\"User\":{}}"}],"SealProof":0,"Activation":0,"Expiration":0,"DealWeight":"0","VerifiedDealWeight":"0","InitialPledge":"0","OnTime":0,"Early":0}}
```

I get 220 of these within 1-2 seconds, and then none for another 20 minutes.

**Comment by damonhung** [comment]
Still seeing this every 2-4 days.  Have been timely upgrading to each new version, latest instance is on v1.1.0.

**Comment by xinbin1122** [comment]
I have the same issue, what's the reason?

**Comment by damonhung** [comment]
@magik6k Really appreciate the fix, but now I'm wondering if it's safe to cherry-pick this change while waiting for the next release.  I'm getting hit with sector fault fees because this bug triggered **three days in a row during window post**:

```2020-11-07T16:25:00.614-0500    WARN    storageminer    storage/wdpost_sched.go:108     window post scheduler notifs channel closed
```
I happened to catch it today during my proving window, restarted `lotus-miner`, and it triggered again within minutes.  But this time, it also did something new: **I lost sync until I stopped the miner.**  I saw the daemon lose sync right when the miner bug happened, and then `lotus sync wait` made no progress for several minutes (1-2 messages / second), then I stopped the miner and the daemon performed re-sync at normal speed.

Miner log uploaded.  It's big, but the error is near the top.  Daemon log available if needed.  Versions:

```~$ lotus-miner version
Daemon:  1.1.2+git.d4cdc6d33.dirty+api0.16.0
Local: lotus-miner version 1.1.2+git.d4cdc6d33.dirty
moo@FIL-SEAL-1:~$ lotus version
Daemon:  1.1.2+git.d4cdc6d33.dirty+api0.17.0
Local: lotus version 1.1.2+git.d4cdc6d33.dirty
```
